src/Solvers/discrete_transforms.jl

```diff
@@ -0,0 +1,178 @@
+abstract type AbstractTransformDirection end
+
+struct Forward <: AbstractTransformDirection end
+struct Backward <: AbstractTransformDirection end
+
+struct DiscreteTransform{P, D, A, G, Δ, Ω, N, T, Σ}
+              plan :: P
+      architecture :: A
+              grid :: G
+         direction :: D
+              dims :: Δ
+          topology :: Ω
+     normalization :: N
+           twiddle :: T
```

Comment:
this is a technical name

Reply:
True. I'll rename to `twiddle_factors` which is still pretty technical but has the benefit of having a Wikipedia page: https://en.wikipedia.org/wiki/Twiddle_factor

Reply:
Wow, I was joking but it _is_ a technical name. Could write

```julia
twiddle_factors :: T # https://en.wikipedia.org/wiki/Twiddle_factor
```